react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+class ValidateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: '',
+      message: 'Password is required'
+    };
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handlePasswordChange(event) {
+    const value = event.target.value;
+
+    if (value === '') {
+      this.setState({ message: 'Password is required' });
+      return;
+    }
+    if (value.length < 8) {
+      this.setState({ message: 'Your password is too short' });
```

Comment:
No matter the outcome you want to still update the `password` property in state using `value`